hack/check-state.sh

```diff
@@ -0,0 +1,210 @@
+     
+    if [ "x$OPERATION_PROGRESSING" != 'xTrue' ] && [ "x$OPERATION_PROGRESSING" != 'xFalse' ]; then
+        echo "Error: OperationProgressing not valid: $OPERATION_PROGRESSING_JSON"
+        exit 1
+    fi
+}
+
+function get_application_degraded_json() {
+    APPLICATION_DEGRADED_JSON=$(echo "$HCO_JSON" | sed -e 's/.*ApplicationDegraded: "\([^"]*\)" "\([^"]*\)" "\([^"]*\)".*/Status: \1 Reason: \2 Message: \3/')   
+    APPLICATION_DEGRADED=$(echo "$HCO_JSON" | sed -e 's/.*ApplicationDegraded: "\([^"]*\)".*/\1/')   
+  
+    if [ "x$APPLICATION_DEGRADED" != 'xTrue' ] && [ "x$APPLICATION_DEGRADED" != 'xFalse' ]; then
+        echo "Error: ApplicationDegraded not valid: $APPLICATION_DEGRADED_JSON"
+        exit 1
+    fi
+}
+
+
+function check_operator_up() {
+   local kind=$1
+   local name=$2
+   local namespace=$3
+
+cat <<EOF
+
+Checking operator kind: $kind name: $name 
+
+EOF
+
+   retry_count=0
+   while [ "$retry_count" -lt "$MAX_APP_AVAILABLE_CHECKS" ]; do
+      get_operator_json $kind $name $namespace 
+    
+      #only check upon first iteration if status attributes are supported
+      if [ "$retry_count" == "0" ]; then
+         has_app_available=`echo "$HCO_JSON" | grep ApplicationAvailable: | wc -l`
+         has_app_degraded=`echo "$HCO_JSON" | grep ApplicationDegraded:  | wc -l`
+         has_app_progressing=`echo "$HCO_JSON" | grep ApplicationDegraded:  | wc -l`
+         has_reconcile=`echo "$HCO_JSON" | grep ReconcileComplete: | wc -l`
+
+         #echo "has_app_available $has_app_available has_app_degraded $has_app_degraded has_app_progressing $has_app_progressing has_reconcile $has_reconcile"
+         
+         if [ "$has_app_available" == "0" ] || [ "$has_app_degraded" == "0" ] || [ "$has_app_progressing" == 0 ]; then
+            echo "Operator $kind doesn't support required status conditions. Skipping check."
+            return
+         fi
+      fi
+  
+      HAS_RECONCILE=""
+      if [ "x$has_reconcile" != "x0" ]; then
+         get_reconcile_json 
+
+         if [ "x$RECONCILE_COMPLETED" == 'xFalse' ]; then
+
+            ((retry_count=retry_count+1))
+            echo "Waiting. Operator $kind :: reconcile not yet complete... (Retry ${retry_count}/${MAX_APP_AVAILABLE_CHECKS}) "
+            sleep 10
+            continue
+         fi
+         HAS_RECONCILE="Reconcile completed and "
+      fi
+
+      get_application_available_json
+      get_application_degraded_json
+      get_operation_progressing_json
+    
+      if [ "x$APPLICATION_AVAILABLE" == 'xTrue' ] && [ "x$APPLICATION_DEGRADED" == 'xFalse' ]; then
+        if [ "x$OPERATION_PROGRESSING" == 'xFalse' ]; then
+          echo "Success: Operator kind: $kind name: $name ${HAS_RECONCILE}is fully available"
+          return
+        fi
+        REASON="Operator $kind available & not degraded, however still progressing"    
+      else
+        if [ "x$OPERATION_PROGRESSING" == 'xFalse' ]; then
+            set +x
```

Comment:
not sure you need this here.

Reply:
I think we need that.
If OperationAvailable==False || OperationDegraded==True then 
     if OperationProgressing==False then 
        error - no more hope for change
     fi

Reply:
I was referring to the `set +x`